Mathlib/Algebra/Order/BigOperators/Ring/Finset.lean

```diff
@@ -208,6 +208,40 @@ theorem sq_sum_div_le_sum_sq_div [LinearOrderedSemifield R] [ExistsAddOfLE R] (s
   rw [div_mul_cancel₀]
   exact (hg i hi).ne'
 
+/-- Weighted **AM-HM inequality**: The weighted harmonic mean is less than or equal to the
+arithmetic mean. -/
+theorem inv_sum_div_le_sum_mul [LinearOrderedSemifield R] [ExistsAddOfLE R] (s : Finset ι)
+    {w f : ι → R} (hw : ∑ i ∈ s, w i = 1) (hw' : ∀ i ∈ s, 0 < w i) (hf : ∀ i ∈ s, 0 < f i) :
+    (∑ i ∈ s, w i / f i)⁻¹ ≤ ∑ i ∈ s, w i * f i := by
+  convert sq_sum_div_le_sum_sq_div s w fun i hi ↦ div_pos (hw' i hi) (hf i hi) using 1
+  · rw [hw, one_pow, one_div]
+  · simp_rw [pow_two, mul_div_assoc]
+    refine sum_congr rfl fun i hi ↦ ?_
+    rw [div_div_cancel' (hw' i hi).ne']
+
+/-- **AM-HM inequality**: The harmonic mean is less than or equal to the arithmetic mean. -/
+theorem div_sum_inv_le_sum_div [LinearOrderedSemifield R] [ExistsAddOfLE R] (s : Finset ι)
+    {f : ι → R} (hf : ∀ i ∈ s, 0 < f i) : #s / ∑ i ∈ s, (f i)⁻¹ ≤ (∑ i ∈ s, f i) / #s := by
```

Comment:
Kind of a shame not using
```suggestion
    {f : ι → R} (hf : ∀ i ∈ s, 0 < f i) : #s / ∑ i ∈ s, (f i)⁻¹ ≤ 𝔼 i ∈ s, f i := by
```
here (but you will need to move the result to a later file to use that spelling). In fact you should probably even do
```suggestion
    {f : ι → R} (hf : ∀ i ∈ s, 0 < f i) : (𝔼 i ∈ s, (f i)⁻¹)⁻¹ ≤ 𝔼 i ∈ s, f i := by
```

Reply:
I think it'd be a bit weird if this file housed the weighted version without the homogeneous version. Maybe we can have both this and a version written with `Finset.expect` in a separate file?